lib/lte_link_control/lte_lc.c

```diff
@@ -1036,98 +1038,108 @@ int lte_lc_psm_get(int *tau, int *active_time)
 	return err;
 }
 
-int lte_lc_edrx_param_set(const char *edrx)
+int lte_lc_edrx_param_set(enum lte_lc_lte_mode mode, const char *edrx)
 {
+	char *edrx_param;
+
+	if (mode != LTE_LC_LTE_MODE_LTEM && mode != LTE_LC_LTE_MODE_NBIOT) {
+		LOG_ERR("LTE mode must be LTE-M or NB-IoT");
+		return -EINVAL;
+	}
+
 	if (edrx != NULL && strlen(edrx) != 4) {
 		return -EINVAL;
 	}
 
-	if (edrx != NULL) {
+	edrx_param = (mode == LTE_LC_LTE_MODE_LTEM) ? edrx_param_ltem :
```

Comment:
Shouldn't this be the other way around. That ```edrx_param_ltem``` and ```edrx_param_nbiot``` gets written to based in incoming ```edrx``` value. The way I read this is that ```edrx_param``` is the only variable that gets manipulated and its only visible to this function.

Reply:
> That edrx_param_ltem and edrx_param_nbiot gets written to based in incoming edrx value

This is what's happening here. `edrx_param` is set to point to either `edrx_param_ltem` or `edrx_param_nbiot` based on the value of `mode`. Then `edrx_param` (that now points to LTE-M or NB-IoT string buffer) is destination for string copy. It's just done this way to reduce code repetition.

Reply:
Yes, of course. Overlooked that part.